### ISSUE TITLE
Endpoint PUT ignores error and executes GET after it

### ISSUE THREAD

**Issue by vekunz** [sveltejs/kit#4210]
### Describe the bug

I have an endpoint with a PUT function. If I return `{status: 401}`, SvelteKit seems to ignore this error, executes the GET endpoint, and the client just gets a 200-response. Therefore my browser-side JavaScript does not know that there was an error, and the user just sees an success message although an error occurred.

### Reproduction

endpoint.ts
```ts
export const get: RequestHandler<Params, any> = async ({ params }) => {
  return {
    body: { foo: "bar" }
  }
}

export const put: RequestHandler<Params, any> = async ({ request, locals }) => {
  return {
    status: 401,
    body: { error: "error" }
  };
}
```

page.svelte
```ts
try {
  await fetch(`/endpoint`, {
    method: 'PUT',
    headers: {
      'Content-Type': 'application/json'
    },
    body: JSON.stringify(content)
  });
} catch (e) {
  alert(JSON.stringify(e));
}
```

### System Info

```shell
System:
    OS: Windows 10 10.0.18363
    CPU: (12) x64 Intel(R) Core(TM) i7-9850H CPU @ 2.60GHz
    Memory: 17.25 GB / 31.79 GB
  Binaries:
    Node: 16.14.0 - C:\Program Files\nodejs\node.EXE
    npm: 8.3.1 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Chrome: 98.0.4758.102
    Edge: Spartan (44.18362.1593.0)
    Internet Explorer: 11.0.18362.1766
  npmPackages:
    @sveltejs/adapter-node: next => 1.0.0-next.70
    @sveltejs/kit: next => 1.0.0-next.288
    svelte: ^3.44.0 => 3.46.4
```

### Severity

blocking all usage of SvelteKit

**Comment by baseballyama** [comment]
Duplicate of https://github.com/sveltejs/kit/issues/3721 ?

**Comment by baseballyama** [comment]
@vekunz 

Hi. I tried to reproduce this issue. but It seems it works properly. (I could get 401 on browser-side)
Could you please check below REPL link and please provide me additional information on how to reproduce it?

https://stackblitz.com/edit/sveltejs-kit-template-default-lp7vor?file=src%2Froutes%2Findex.svelte&terminal=dev

**Comment by vekunz** [comment]
Hi, I now figured out, where the problem is. We use a page endpoint and manually invoke the PUT call (and not in the load function). And if I just send "Content-Type: application/json" in the fetch, SvelteKit first executes the PUT-Endpoint and then returns the rendered HTML (like GET with text/html). But if I set "Content-Type: application/json" and "Accept: application/json", it works fine.

I think the actual problem here is, that a PUT-Call with just "Content-Type: application/json" both executes the Endpoint and returns the pre-rendered HTML.

I modified you Stackblitz example for this case: https://stackblitz.com/edit/sveltejs-kit-template-default-wagmb5?file=src%2Froutes%2Fendpoint.svelte&terminal=dev
